exp/api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -363,3 +383,21 @@ func validateBoolPtrImmutable(path *field.Path, oldVal, newVal *bool) *field.Err
 
 	return nil
 }
+
+func validateStringPtrImmutable(path *field.Path, oldVal, newVal *string) *field.Error {
```

Comment:
Does this perhaps belong as something like `util/strings/strings.go` util package like we do for maps & slices so that it can be reused by other validation webhooks?

Reply:
That sounds like a good idea to me. @jackfrancis do you have any opinions on how that might relate to #2718?

Reply:
+1, my vote would be to fold the effort of generalizing this foo into a common library into #2718 so we can land this change.

Also we want to have func-specific UT at some point, I can take that on in #2718 as well.